guides/k8-akka-cluster/src/main/paradox/index.md

```diff
@@ -0,0 +1,455 @@
+
+
+Either one of the following code can be used by your application to set up the `ActorSystem` name.
+
+```
+val actorSystem = sys.props.get("actorSystemName")
+  .fold(throw new IllegalArgumentException("Actor system name must be defined"))(ActorSystem(_))
+```
+
+Or alternatively, if you have a custom configuration.
+
+```
+import com.typesafe.config.ConfigFactory
+
+val actorSystem = sys.props.get("actorSystemName")
+  .fold(throw new IllegalArgumentException("Actor system name must be defined")) { actorSystemName =>
+    val config = ConfigFactory.load("custom.conf")
+    val actorSystem = ActorSystem(actorSystemName, config)
+  }
+```
+
+### Publishing to Docker registry - SBT
+
+We will be using [SBT Native Packager](http://www.scala-sbt.org/sbt-native-packager/) plugin to containerize the application.
+
+Enable SBT Native Packager in your project by adding the following line in the `project/plugins.sbt`.
+
+```
+addSbtPlugin("com.typesafe.sbt" % "sbt-native-packager" % "1.2.0")
+```
+
+Enable the [JavaAppPackaging](http://www.scala-sbt.org/sbt-native-packager/archetypes/java_app/index.html) plugin provided by SBT Native Packager.
+
+```
+enablePlugins(JavaAppPackaging)
+```
+
+Alternatively you may choose to enable [JavaServerAppPackaging](http://www.scala-sbt.org/sbt-native-packager/archetypes/java_server/index.html) instead. The `JavaServerAppPackaging` plugin provides all the features provided by the `JavaAppPackaging` plugin with some additional [server features](http://www.scala-sbt.org/sbt-native-packager/archetypes/java_server/index.html#features).
+
+```
+enablePlugins(JavaServerAppPackaging)
+```
+
+Please note that you must choose to enable either `JavaAppPackaging` or `JavaServerAppPackaging` - you _can't_ enable both.
+
+We will now configure the Docker settings required to containerize the application.
+
+Append the Akka clustering related sytem properties to the `dockerEntrypoint` setting.
+
+```
+dockerEntrypoint ++= """-Dplay.crypto.secret="${APPLICATION_SECRET:-none}" -Dplay.akka.actor-system="${AKKA_ACTOR_SYSTEM_NAME:-friendservice-v1}" -Dhttp.address="$FRIENDSERVICE_BIND_IP" -Dhttp.port="$FRIENDSERVICE_BIND_PORT" -Dakka.actor.provider=cluster -Dakka.remote.netty.tcp.hostname="$AKKA_REMOTING_BIND_HOST" -Dakka.remote.netty.tcp.port="$AKKA_REMOTING_BIND_PORT" -Dakka.cluster.seed-nodes.0="akka.tcp://${AKKA_ACTOR_SYSTEM_NAME}@${AKKA_SEED_NODE_HOST}:${AKKA_SEED_NODE_PORT}" -Dakka.io.dns.resolver=async-dns -Dakka.io.dns.async-dns.resolve-srv=true -Dakka.io.dns.async-dns.resolv-conf=on""".split(" ").toSeq
+```
+
+As part of building the Docker image, SBT Native Packager will provide its own Docker entry point script to start the application. which accepts optional additional arguments. When system properties are presented as part of the arguments, they will be appended to the JVM options when the application is started within the container.
+
+Next we will transform the generated Docker `ENTRYPOINT` instruction.
```

Comment:
Describe where to add the `dockerCommands` setting, e.g. "Next, we will transform the generated Docker `ENTRYPOINT` instruction in the `build.sbt`."